sequencer/src/state.rs

```diff
@@ -189,37 +262,68 @@ impl HotShotState<SeqTypes> for ValidatedState {
     /// proposal descends from parent. Returns updated `ValidatedState`.
     async fn validate_and_apply_header(
         &self,
-        _instance: &Self::Instance,
+        instance: &Self::Instance,
         parent_leaf: &Leaf,
         proposed_header: &Header,
     ) -> Result<Self, Self::Error> {
+        // Fetch the new L1 deposits between parent and current finalized L1 block.
+        let l1_deposits = fetch_fee_receipts(
+            parent_leaf.get_block_header().l1_finalized,
+            proposed_header.l1_finalized,
+        );
+
+        // Find missing state entries
+        let missing_accounts = self.forgotten_accounts(
+            std::iter::once(proposed_header.fee_info.account)
+                .chain(l1_deposits.iter().map(|fee_info| fee_info.account)),
+        );
+
+        let view = parent_leaf.get_view_number();
+
         // Clone state to avoid mutation. Consumer can take update
         // through returned value.
         let mut validated_state = self.clone();
```

Comment:
Nit: I like keeping this at the top just for emphasis. Its something that is obvious once you have familiarity w/ the code, but for the unfamiliar can clarifies somewhat how the caller will interact w/ our code.